neuro_flow/context.py

```diff
@@ -1225,10 +1224,10 @@ async def create(  # type:ignore[override]
         return cast(_CtxT, replace(ctx, _prefix=prefix, _prep_tasks=prep_tasks))
 
     async def calc_outputs(self, needs: NeedsCtx) -> DepCtx:
-        if any(i.result == JobStatus.CANCELLED for i in needs.values()):
-            return DepCtx(JobStatus.CANCELLED, {})
-        elif any(i.result == JobStatus.FAILED for i in needs.values()):
-            return DepCtx(JobStatus.FAILED, {})
+        if any(i.result == TaskStatus.CANCELLED for i in needs.values()):
```

Comment:
I have a design question: should we mark action as disabled if any of its tasks were disabled?

Reply:
BatchAction -- yes, probably

Reply:
OK, but I think it can be surprising that batch action did not produced any results because some intermediate task was skipped.

Reply:
I think we should probably allow to specify `needs` for batch action outputs. But I agree that it sounds a little be weird.

Reply:
You can use `needs` for accessing action's outputs. This is the legit way.